libs/platform/src/lib/components/form/select/select/select.component.ts

```diff
@@ -0,0 +1,254 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    Host,
+    OnInit,
+    Optional,
+    Self,
+    SkipSelf,
+    ViewChild,
+    ViewEncapsulation,
+    Input,
+    EventEmitter,
+    Output,
+    AfterViewChecked
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+
+
+import { SelectComponent as CoreSelect } from '@fundamental-ngx/core';
+
+import { DynamicComponentService, FdSelectChange, SelectControlState } from '@fundamental-ngx/core';
+import { OptionItem } from '../../../../domain';
+import { FormField } from '../../form-field';
+import { FormFieldControl } from '../../form-control';
+import { BaseSelect, FdpSelectionChangeEvent } from '../commons/base-select';
+import { SelectConfig } from '../select.config';
+
+@Component({
+    selector: 'fdp-select',
+    templateUrl: './select.component.html',
+    styleUrls: ['./select.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [{ provide: FormFieldControl, useExisting: SelectComponent, multi: true }]
+})
+export class SelectComponent extends BaseSelect implements OnInit, AfterViewInit, AfterViewChecked {
+
+    /** Whether the select component is disabled. */
+    @Input()
+    state: SelectControlState = null;
+
+    /**
+    * Directly sets value to the component that at the ends up at writeValue as well fires
+    * change detections
+    *
+    */
+    @Input()
+    get value(): any {
+        return this._value;
+    }
+
+    set value(newValue: any) {
+        if (newValue !== this._value) {
+            this._value = newValue;
+            this.onChange(newValue);
+            this.onTouched();
+            this.valueChange.emit(newValue);
+            this.cd.markForCheck();
+        }
+    }
+
+    /** Event emitted when the selected value of the select changes. */
+    @Output()
+    readonly valueChange: EventEmitter<FdSelectChange> = new EventEmitter<FdSelectChange>();
+
+    @ViewChild(CoreSelect, { static: true })
+    select: CoreSelect;
+
+    /** @hidden */
+    selected?: OptionItem;
+
+    constructor(
+        readonly cd: ChangeDetectorRef,
+        readonly elementRef: ElementRef,
+        @Optional() @Self() readonly ngControl: NgControl,
+        @Optional() @Self() readonly ngForm: NgForm,
+        readonly _dynamicComponentService: DynamicComponentService,
+        readonly _selectConfig: SelectConfig,
+        @Optional() @SkipSelf() @Host() formField: FormField,
+        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>
+    ) {
+        super(cd, elementRef, ngControl, ngForm, _selectConfig, formField, formControl);
+    }
+
+    /** @hidden
+     */
+    ngOnInit(): void {}
```

Comment:
please remove empty state

Reply:
removal of it leads to compile time error as =>"at SelectComponent.push.FWTC.BaseInput.ngOnInit"